### PR TITLE
chore: reorganize harness tests

### PR DESCRIPTION
**Pull request by hal3e**
closes: https://github.com/FuelLabs/fuels-rs/issues/896

In this PR, I propose a new harness tests layout for types. The main idea is to separate types tests based on the program. Initially, we agreed on `contracts` and `scripts` but I have also separated `predicates`. This is the new structure:

```shell
tests
...
│
├── types/
│  ├── contracts/
│  │  ├── b512/
│  │  ...
│  │  ├── vector_output/
│  │  └── vectors/
│  ├── predicates/
│  │  ├── address/
│  │  ...
│  │  ├── structs/
│  │  └── u64/
│  └── scripts/
│     ├── options_results/
│     │  ...
│     └── script_vectors/
 ...
├── types_contracts.rs
├── types_predicates.rs
└── types_scripts.rs
...
```
Changes:
- `types` folder has 3 sub-folder for each program
- `types.rs` was split into three files `types_contracts.rs`, `types_predicates.rs` and `types_scripts.rs`
- Removed program name from sway projects where they were not needed. E.g.  `contract_storage_test` -> `storage`
- Deleted duplicated test: `type_safe_output_values`
- Deleted unused sway projects: `require_contract_abi` and `require_from_contract` and `script_vector`

### Checklist
- [x] I have linked to any relevant issues.
- [x] I have updated the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: packages/fuels/tests/types_contracts.rs
Comment (iqdecay) [inline packages/fuels/tests/types_contracts.rs:64]: This was deleted because it is redundant?

**Review by iqdecay — CHANGES_REQUESTED** [review]
Just one question

**Comment by iqdecay** [comment]
I think we hold off merging until #815 is merged?

**Comment by hal3e** [comment]
> I think we hold off merging until #815 is merged?

Makes sense. That PR is done. We just need to merge it.

**Review by iqdecay — DISMISSED** [review]
noice

**Review by iqdecay — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
Nice -- clean re-organization!

**Pull request merged**
